### PR TITLE
Feature/1100 extra custom scalar where fields

### LINKED ISSUE #1100: Filtering: WHERE type wont get autogenerated on custom scalars
**Describe the bug**
This is probably heavily correlated to #915.
I´m still converting my old neo4j-graphql-js queries to V3.0.1 and I need the custom scalar UnsignedInt from 'graphql-scalars'.

**Type definitions**
```
type Material {
  Number: UnsignedInt! @unique
  Name: String!
}
```

**To Reproduce**

1. Run a server with the the typedef above

**Expected behavior**
The autogenerated MaterialWhere only generates _IN, _NOT_IN, etc. for the Name field.
This should also be the case for any custom scalars.

**Screenshots**
image

**System:**

* OS: Windows
* Version: @neo4j/graphql@3.0.1
* Node.js version: 16.6.1

**Additional context**
Maybe this can be fixed with #915 all together.

### PR DESCRIPTION
**Pull request by Liam-Doodson**
# Description

Added the following extra where fields for custom scalars:
* For a list field: _NOT, _INCLUDES and _NOT_INCLUDES
* For a singular field: _NOT, _IN and _NOT_IN

Updated schema tests to check for these extra fields.
Added integration tests to test filtering with the extra fields.

# Issue

Closes #1100 

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [ ] TCK tests have been updated
- [ ] Integration tests have been updated
- [ ] Example applications have been updated
- [ ] New files have copyright header
- [ ] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at
 http://neo4j-graphql-toolbox-stage-2175.surge.sh (=> for local development; use for instance [Neo4j Desktop](https://neo4j.com/download/))
or
 https://neo4j-graphql-toolbox-stage-2175.surge.sh (=> use to connect to an [Aura Free](https://neo4j.com/cloud/aura-free/) dbms)

**Review by darrellwarde — CHANGES_REQUESTED** [review]
One change that is definitely needed is the addition of a changeset for this work - you do this by going to the root of your cloned repo and running `yarn changeset`, which will launch a wizard to walk you through adding a changelog entry and specifying the version. This should be within the development documentation, that's on us.

Other than that, some general feedback is you have added lots of tests (which is great!), however, they are all for positive cases where the result matches the filter. It would be good to see some tests where there are zero expected results which match the filter, in case for some reason the result is _always_ returned. But this isn't a blocker!

Nice work, you figured out where to add this in quickly. 🙂

File: packages/graphql/tests/integration/custom-scalar-filtering.int.test.ts [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+import { generateUniqueType } from "../utils/graphql-types";
+
+describe("Custom Scalar Filtering", () => {
+    let driver: Driver;
+    let neo4j: Neo4j;
+
+    beforeAll(async () => {
+        neo4j = new Neo4j();
+        driver = await neo4j.getDriver();
+        process.env.NEO4J_GRAPHQL_ENABLE_REGEX = "true";
```
[the remark this answers is not in the document]
Comment (Liam-Doodson) [reply packages/graphql/tests/integration/custom-scalar-filtering.int.test.ts]: Ah yes I meant to delete those lines!

File: .changeset/thick-boats-peel.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@neo4j/graphql": minor
+---
+
+Added extra where fields for custom scalars. For lists of custom scalars \_NOT, \_INCLUDES and \_NOT_INCLUDES are now supported. For a single value custom scalars \_NOT, \_IN and \_NOT_IN are now supported.
```
Comment (darrellwarde) [inline .changeset/thick-boats-peel.md:5]: Excellent changelog entry! 👏

**Review by darrellwarde — APPROVED** [review]
All looks good to me, nice work!

File: .changeset/thick-boats-peel.md (hunk 2 above)
Comment (Liam-Doodson) [reply .changeset/thick-boats-peel.md:5]: Thank you!

**Pull request merged**